### PR TITLE
(DOCSP-22069): offline login - Node.js

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-22069

### Staged Changes (Requires MongoDB Corp SSO)

- [Open and Close a Realm - Node.js](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22069-offline-login/sdk/node/examples/open-and-close-a-realm/)
- [Authenticate Users - Node.js](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22069-offline-login/sdk/node/examples/authenticate-users/)

- [Open and Close a Realm - React Native](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22069-offline-login/sdk/react-native/examples/open-and-close-a-realm/)
- [Authenticate Users - React Native](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-22069-offline-login/sdk/react-native/examples/authenticate-users/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/node/examples/authenticate-users.txt [hunk 1]
```diff
@@ -342,6 +342,15 @@ use to finish logging the user in to your app.
 
 .. include:: /includes/authorization-appleidcredential-string.rst
 
+.. _node-offline-login:
+
+Offline Login
+-------------
+
+To learn how to use a synced realm while offline or with unstable internet
```
Comment (dacharyc) [inline source/sdk/node/examples/authenticate-users.txt:350]: I think this page would benefit from including the authentication code snippet here - i.e. [/examples/generated/node/open-and-close-a-realm.codeblock.use-cached-user-to-login.js](https://github.com/mongodb/docs-realm/pull/1918/files#diff-43e85a25480467f696368102c722db920feefee2b5bd34a1db7b35493b6c87d4) - and then saying something like "For additional information about how to work with realms when offline or with an unstable internet connection, see..."

This page is really about the authentication piece, so the nuances of whether to download before open or use a timeout when opening aren't really applicable here. All of the other examples on this page are about the authentication piece - none of them go into details about how to open a realm after authenticating. This separation would give devs the basic example of how to handle authentication while a user is offline where they would expect to see it, but direct them to the "open a realm" page for all those other details.

We could probably even remove it from the other page, since the other page is about opening a realm - not authenticating.

**Review by dacharyc — APPROVED** [review]
Nice work! This _could_ be merged as-is, but I think there are a few improvements we could make to make this a little more useful (and readable) for devs.

File: source/sdk/node/examples/open-and-close-a-realm.txt [hunk 2]
```diff
@@ -45,115 +58,77 @@ In the SyncConfiguration, you must include include a ``user`` and ``flexible:tru
...
 
-To open an existing realm with a different configuration, pass the new 
-:js-sdk:`Configuration <Realm.html#~Configuration>`
-to :js-sdk:`Realm.writeCopyTo() <Realm.html#writeCopyTo>`. 
-``Realm.writeCopyTo()`` copies the realm's data to a new realm, but adds different configuration options. 
-In the output realm's configuration, you *must* specify the ``path``. 
+To copy data from an existing {+realm+} to a new realm with different
+configuration options, you can use the :js-sdk:`Realm.writeCopyTo() <Realm.html#writeCopyTo>` method.
+
+In the new realm's configuration, you *must* specify the ``path``.
```
Comment (mongodben) [inline source/sdk/node/examples/open-and-close-a-realm.txt:131]: why this change?

File: source/sdk/node/examples/authenticate-users.txt [hunk 3]
```diff
@@ -342,6 +342,15 @@ use to finish logging the user in to your app.
...
+connection, read the :ref:`Open a Synced Realm While Offline
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/node/examples/authenticate-users.txt:351]: Going to stick to the o.g wording I had for this since I personally prefer it but I agree that both are equally acceptable

File: source/sdk/node/examples/authenticate-users.txt (hunk 1 above)
Comment (mongomoe) [reply source/sdk/node/examples/authenticate-users.txt:360]: Fair point on adding that getUser example to the auth page, I decided to keep it in the open a realm page as well since I feel like the getUser example flows nicely into the two background sync examples.

File: source/sdk/node/examples/open-and-close-a-realm.txt [hunk 4]
```diff
@@ -45,115 +58,77 @@ In the SyncConfiguration, you must include include a ``user`` and ``flexible:tru
...
-By default, {+service-short+} syncs all data from the server before returning. 
+.. literalinclude:: /examples/generated/node/open-and-close-a-realm.codeblock.use-cached-user-to-login.js
+   :language: javascript
 
-.. example:: 
+The following subsections show how to open a {+realm+} using the cached user,
+and :js-sdk:`OpenRealmBehaviorConfiguration
+<Realm.App.Sync.html#~OpenRealmBehaviorConfiguration>`. The
+``OpenRealmBehaviorConfiguration`` allows you to specify configuration options
+for background synchronization. The ``newRealmFileBehavior`` and
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/node/examples/open-and-close-a-realm.txt:82]: On second look, I agree that the initial wording was hard to follow, but I feel your suggestion lost what the details being specified were exactly (i.e background sync details). So instead I rephrased it to:

```
The following subsections show how to use background synchronization to access a
realm while offline. To do this, use the cached user and an
OpenRealmBehaviorConfiguration object. 

Within your Sync Configuration, set the optional newRealmFileBehavior and
existingRealmFileBehavior fields to your OpenRealmBehaviorConfiguration object
to enable background synchronization. 
```

File: source/sdk/node/examples/open-and-close-a-realm.txt [hunk 5]
```diff
@@ -45,115 +58,77 @@ In the SyncConfiguration, you must include include a ``user`` and ``flexible:tru
...
+.. _node-open-immediately-with-background-sync:
+
+Open Immediately with Background Sync
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   The following example shows how to open a synced {+realm+} with with a 
-   ``SyncConfiguration`` object that uses a predefined
-   ``CarSchema`` :doc:`schema </sdk/node/examples/define-a-realm-object-model>`, 
-   the currently logged in user, and a partition value of "myPartition".
+If the user's device is not connected to the internet or you're uncertain of
```
[the remark this answers is not in the document]
Comment (mongomoe) [reply source/sdk/node/examples/open-and-close-a-realm.txt:97]: Made it a bit of a hybrid of both:

```
If the user's device is not connected to the internet or you're uncertain of
it's connection status, set the realm behavior's type to ``openImmediately``. This 
syncs data from the server in the background. 
```

**Comment by mongomoe** [comment]
Fyi I merged this despite issues w/ the Node Testing suite since those issues are related to mongodb.ts and I didn't touch that file in this pr. Will look into those issues separately.
